compiler/passes/codegen.cpp

```diff
@@ -607,11 +602,13 @@ static void codegen_header_compilation_config() {
     fprintf(cfgfile.fptr,
             "printf(\"%%s\", \"  CHPL_HOME: %s\\n\");\n",
             CHPL_HOME);
-    for (int i = 0; i < num_chpl_env_vars; i++) {
-      fprintf(cfgfile.fptr,
-              "printf(\"%%s\", \"  %s: %s\\n\");\n",
-              chpl_env_var_names[i],
-              chpl_env_vars[i]);
+    for (std::map<std::string, const char*>::iterator env=envMap.begin(); env!=envMap.end(); ++env) {
```

Comment:
Huh, I was a little surprised that we don't have a `for_map` macro like we do for other std data types. Nothing you have to do for this PR, it was just surprising. @Kyle-B or @noakesmichael can you think of any reason we don't have a `for_map` in [stlUtil.h] (https://github.com/ben-albrecht/chapel/blob/newcompflags/compiler/include/stlUtil.h#L52-L62)?